stdlib/asm/crypto/stark/deep_queries.masm

```diff
@@ -1,245 +1,471 @@
 use.std::crypto::stark::constants
 
-#! Computes a single step of the random linear combination defining the DEEP composition polynomial
-#! that is the input to the FRI protocol. More precisely, the sum in question is:
-#! $$
-#! \sum_{i=0}^k{\alpha_i \cdot \left(\frac{T_i(x) - T_i(z)}{x - z} +
-#!            \frac{T_i(x) - T_i(z \cdot g)}{x - z \cdot g} \right)}
-#! $$
+#! Computes the adjustment factors needed during Horner evaluation.
 #!
-#! and the following instruction computes the denominators $\alpha_i \cdot (T_i(x) - T_i(z))$ and
-#! $\alpha_i \cdot (T_i(x) - T_i(z \cdot g))$ and stores the values in two accumulators $r$ and $p$,
-#! respectively. This instruction is specialized to auxiliary trace columns i.e. the values $T_i(x)$
-#! are field elements in a quadratic extension field.
+#! As `horner_eval_base` and `horner_eval_ext` operate on batches of 8 and 4 coefficients,
+#! respectively, and when the number of coefficients of the polynomial we are evaluating
+#! is not divisible by the size of the batch, we need to adjust the result to account
+#! for this. The adjustment is performed by multiplying the Horner accumulator by
+#! alpha^i where i depends on the number of coefficients missing in the last processed batch.
+#! For example, if we are evaluating using `horner_eval_base`, and the last batch of coefficients
+#! has only 7 coefficients, we would need then to multiply by alpha^1 to adjust the result.
+#! Note that since we are evaluating throughout at alpha^{-1}, the adjustment factors are
+#! alpha^i, if on the other hand we were evaluating at alpha then the factors would be alpha^{-i}.
 #!
-#! The stack transition of the instruction can be visualized as follows:
+#! Input: [...]
+#! Output: [...]  
 #!
-#! +-------+-------+-------+-------+-------+-------+-------+-------+------+------+------+------+------+------+------+---+
-#! |  T31  |  T30  |  T21  |  T20  |  T11  |  T10  |  T01  |  T00  |  p1  |  p0  |  r1  |  r0  |x_addr|z_addr|a_addr| - |
-#! +-------+-------+-------+-------+-------+-------+-------+-------+------+------+------+------+------+------+------+---+
+#! Cycles: 146
+proc.compute_adjustment_power_z_quotients
```

Comment:
In the same spirit as the previous comment, we could run this code in `random_coin::generate_deep_composition_random_coefficients`, so that it becomes clearer when it's legal to use the `alpha^i` constants. That is, as soon we have `alpha`, we compute all derived variables - `alpha^i`, `alpha_inv`, and any other